llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -3157,6 +3157,20 @@ bool CombinerHelper::matchHoistLogicOpWithSameOpcodeHands(
     // Match: logic (ext X), (ext Y) --> ext (logic X, Y)
     break;
   }
+  case TargetOpcode::G_TRUNC: {
+    // Match: logic (trunc X), (trunc Y) -> trunc (logic X, Y)
+    MachineFunction *MF = MI.getMF();
+    EVT DstEVT = getApproximateEVTForLLT(MRI.getType(Dst), MF->getDataLayout(),
+                                         MF->getFunction().getContext());
+    EVT XEVT = getApproximateEVTForLLT(XTy, MF->getDataLayout(),
+                                       MF->getFunction().getContext());
```

Comment:
We already have an LLT native isTruncateFree/isZExtFree, you don't need to figure out the LLT here

Reply:
Okay, done.